src/test/java/mate/academy/bookshop/service/impl/AuthenticationServiceImplTest.java

```diff
@@ -0,0 +1,170 @@
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@ExtendWith(MockitoExtension.class)
+class AuthenticationServiceImplTest {
+
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private AuthenticationManager authenticationManager;
+    @Mock
+    private PasswordEncoder passwordEncoder;
+    @Mock
+    private ShoppingCartRepository shoppingCartRepository;
+    @Mock
+    private RoleRepository roleRepository;
+    @Mock
+    private UserMapper userMapper;
+    @Mock
+    private JwtUtil jwtUtil;
+
+    @InjectMocks
+    private AuthenticationServiceImpl authenticationService;
+
+    private UserRegistrationRequestDto requestDto;
+    private User user;
+    private UserResponseDto userResponseDto;
+    private ShoppingCart shoppingCart;
+
+    @BeforeEach
+    void setUp() {
+        requestDto = new UserRegistrationRequestDto();
+        requestDto.setEmail("user@example.com");
+        requestDto.setPassword("password");
+
+        user = new User();
+        user.setEmail(requestDto.getEmail());
+        user.setPassword(requestDto.getPassword());
+
+        userResponseDto = new UserResponseDto();
+        userResponseDto.setEmail(user.getEmail());
+
+        shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+    }
+
+    @DisplayName("Register new user with valid credentials")
+    @Test
+    void register_UserWithUniqueEmail_returnsUserResponseDto() throws RegistrationException {
+        when(userRepository.existsByEmail(requestDto.getEmail())).thenReturn(false);
+        when(userMapper.toModel(requestDto)).thenReturn(user);
+        when(passwordEncoder.encode(user.getPassword())).thenReturn("encodedPassword");
+        when(roleRepository.findByName(RoleName.ROLE_USER)).thenReturn(Set.of(new Role()));
+        when(userRepository.save(user)).thenReturn(user);
+        when(userMapper.toDto(user)).thenReturn(userResponseDto);
+
+        UserResponseDto response = authenticationService.register(requestDto);
+
+        assertEquals(userResponseDto, response);
+    }
+
+    @DisplayName("Register new user with invalid credentials")
+    @Test
+    void register_UserWithNotUniqueEmail_ThrowsRegistrationException()
+            throws RegistrationException {
+        when(userRepository.existsByEmail(requestDto.getEmail())).thenReturn(true);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, ()
+                -> authenticationService.register(requestDto));
+
+        String expected = "There is another user with email " + requestDto.getEmail();
+        String actual = exception.getMessage();
+        assertEquals(expected, actual);
```

Comment:
Try to keep struture in all tests: Given (all mocks, preconditions, setups), When (execution of code under test), Then (assertions and verify) these parts should be separated by empty line to increase readability